eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -122,7 +122,7 @@ object NodeRelay {
   def computeRouteParams(nodeParams: NodeParams, amountIn: MilliSatoshi, expiryIn: CltvExpiry, amountOut: MilliSatoshi, expiryOut: CltvExpiry): RouteParams = {
     val routeMaxCltv = expiryIn - expiryOut
     val routeMaxFee = amountIn - amountOut
-    RouteCalculation.getDefaultRouteParams(nodeParams.routerConf.pathFindingConf).copy(
+    RouteCalculation.getDefaultRouteParams(nodeParams.routerConf.pathFindingExperimentConf.get()).copy(
```

Comment:
The way Phoenix works is that it will make several retries with increased fees if no route can be found. I think it make sense to use the same algo parameters for the subsequent retries, otherwise interpretation might be difficult.

We could derive the experiment from the `payment_hash`, or a combination of `payment_hash` and `payment secret`, if we want separate payment attempts (initiated by the user) to use different parameters.

Reply:
We can add the experiment name in relevant parts of the logs to help debugging. I don't think that using different experiments for subsequent retries is problematic. I'm afraid that always forcing the same experiment for a given payment hash may distort the metrics if someone tries the same payment many times and it keeps failing, if we choose the experiment randomly each time then the effect of such a payment would be spread to all experiments.

Reply:
@t-bast WDYT using log context to track experiments? Never debugged the algo so not sure what is more convenient

Reply:
I think it would be hard to ensure we use the same experiment for retries, we'd need some state in the `NodeRelayer` which I'd like to avoid...

I think it's not an issue that the retries use a different experiment, as long as our events contain the "fee budget" allocated to the path-finding. This way we can compare experiments that have approximately the same fee budget (it indeed doesn't make much sense to compare two events if the fees allocated to the path-finding were widely different).

Reply:
Closing this thread, follow up on "fee budget" is https://github.com/ACINQ/eclair/pull/1930/files#r704265635.